homeassistant/components/sensor/daikin.py

```diff
@@ -0,0 +1,98 @@
+"""
+Support for Daikin AC Sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.daikin/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, CONF_ICON,
+    CONF_MONITORED_CONDITIONS, CONF_TEMPERATURE_UNIT
+)
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components.daikin import (
+    SENSOR_TYPES,
+    manual_device_setup
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=SENSOR_TYPES.keys()):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Daikin sensors."""
+    if discovery_info is not None:
+        host = discovery_info.get('ip')
+        name = None
+        monitored_conditions = discovery_info.get(
+            CONF_MONITORED_CONDITIONS, list(SENSOR_TYPES.keys())
+        )
+        _LOGGER.info("Discovered a Daikin AC sensor on %s", host)
+    else:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+        monitored_conditions = config.get(CONF_MONITORED_CONDITIONS)
+        _LOGGER.info("Added Daikin AC sensor on %s", host)
+
+    device = manual_device_setup(hass, host, name)
+
+    sensors = []
+    for monitored_state in monitored_conditions:
+        sensors.append(DaikinClimateSensor(device, name, monitored_state))
+
+    add_devices(sensors, True)
+
+
+class DaikinClimateSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, device, name=None, monitored_state=SENSOR_TYPES.keys()):
+        """Initialize the sensor."""
+        self._device = device
+        if name is None:
+            name = SENSOR_TYPES.get(monitored_state).get(CONF_NAME) \
+                + ' ' + device.name
+
+        self._name = name
+        self._state = monitored_state
```

Comment:
If I see this correctly, you set `_state` not actually to the state of the sensor, but to the name of the attribute that you want to retrieve from the device, right? In this case, please rename this variable for consistency. Everywhere else that I have seen a `self._state`, it stores the state of the entity.

Reply:
Indeed.
Would _device_attribute be ok or do you have other suggestion for the variable name ?

Reply:
Renamed